### ISSUE TITLE
NuGet sign command should support custom certificate stores

### ISSUE THREAD

**Issue by lemccomb** [NuGet/Home#13373]
### NuGet Product(s) Involved

NuGet.exe

### The Elevator Pitch

Loading the My:CurrentUser certificate store can be a performance bottleneck for signing operations, so it is often a better practice to use a custom certificate store, especially in build systems and other contexts that make a lot of signing calls.
The NuGet.exe sign command should enable this pattern by allowing the user to sign with a certificate in a custom store.

### Additional Context and Details

Currently, running the sign command with -CertificateStoreName set to a custom store name returns the following error:
Invalid value provided for 'CertificateStoreName'

**Comment by zivkan** [comment]
The current implementation (code search found dotnet nuget sign before nuget.exe sign) that we're trying to parse the value into a `StoreName` enum value: https://github.com/NuGet/NuGet.Client/blob/7ad6fcc9c56c960975c37b2416c7eae1d53ba3fd/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/SignCommand.cs#L192-L209

However, X509Store's constructor has an overload with a string name: https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509store.-ctor?view=net-8.0#system-security-cryptography-x509certificates-x509store-ctor(system-string)

Therefore, I assume it's technically feasible, even if I couldn't figure out how to let certmanager to create a custom store.

**Comment by lemccomb** [comment]
Yeah, that's what I figured, and that's how we were doing it in the Windows build signing code as well. But we run >100k signing operations/day, so the perf hit from reading the entire My store was adding up. So now I'm moving our build sign code to use a custom cert store and NuGet is the last hold-out.

For now, as a work-around, I'm copying the NuGet cert into the My store just before signing but it means checking the store for the cert for every package, so still not idea from a perf standpoint.

Thanks for the quick reply,
Leith


Sent from Outlook<http://aka.ms/weboutlook>

________________________________
